pkg/controller/node/nodecontroller.go

```diff
@@ -573,6 +668,42 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+// evictIntolerablePods checks for given node if there's any pod that
+// doesn't tolerate NoExecute taints and send it for eviction.
+func (nc *NodeController) evictIntolerablePods(node *api.Node) (bool, error) {
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if len(taints) == 0 {
+		return false, nil
+	}
+
+	selector := fields.OneTermEqualSelector(api.PodHostField, node.Name)
+	options := api.ListOptions{FieldSelector: selector}
+	pods, err := nc.kubeClient.Core().Pods(api.NamespaceAll).List(options)
+	if err != nil {
+		return false, err
+	}
+
+	evicted := false
+	for _, pod := range pods.Items {
+		tolerations, err := api.GetTolerationsFromPodAnnotations(pod.Annotations)
```

Comment:
Same as above - please create a helper function that doesn't have "Annotations" in it's name.